evalml/problem_types/utils.py

```diff
@@ -19,3 +23,31 @@ def handle_problem_types(problem_type):
     if isinstance(problem_type, ProblemTypes):
         return problem_type
     raise ValueError('`handle_problem_types` was not passed a str or ProblemTypes object')
+
+
+def detect_problem_type(y):
+    """Determine the type of problem is being solved based on the targets (binary vs multiclass classification, regression)
+        Ignores missing and null data
+
+    Arguments:
+        y (pd.Series): the target labels to predict
+
+    Returns:
+        ProblemType: ProblemType Enum
+
+    Example:
+        >>> y = pd.Series([0, 1, 0, 0, 0, 1, 1, 0, 0, 1, 1])
+        >>> problem_type = detect_problem_type(y)
+        >>> assert problem_type == ProblemTypes.BINARY
+    """
+    y = pd.Series(y).dropna()
+    num_classes = y.nunique()
+    if num_classes < 2:
+        raise ValueError("Less than 2 classes detected! Target unusable for modeling")
+    elif num_classes == 2:
+        return ProblemTypes.BINARY
+    else:
```

Comment:
Nit-pick: delete this `else`, not necessary since you `raise`/`return` in the `if`/`elif`:
```python
if num_classes < 2:
    raise ValueError("Less than 2 classes detected! Target unusable for modeling")
if num_classes == 2:
    return ProblemTypes.BINARY
if y.dtype in numeric_dtypes:
    return ProblemTypes.REGRESSION
return ProblemTypes.MULTICLASS
```

This keeps the control flow flat and easy to follow!